src/main/java/org/folio/services/instance/InstanceService.java

```diff
@@ -150,14 +157,39 @@ public Future<Response> updateInstance(String id, Instance newInstance) {
       })
       .compose(oldInstance -> {
         final Promise<Response> putResult = promise();
+        linkOrUnlinkSubjects(newInstance, oldInstance);
         put(INSTANCE_TABLE, newInstance, id, okapiHeaders, vertxContext,
           InstanceStorage.PutInstanceStorageInstancesByInstanceIdResponse.class, putResult);
-
         return putResult.future()
           .onSuccess(domainEventPublisher.publishUpdated(oldInstance));
       });
   }
 
+  private void linkOrUnlinkSubjects(Instance newInstance, Instance oldInstance) {
+    var instanceId = newInstance.getId();
+
+    if (newInstance.getSubjects().isEmpty()) {
+      instanceRepository.unlinkInstanceFromSubjectSource(instanceId);
+      instanceRepository.unlinkInstanceFromSubjectType(instanceId);
+      return;
+    }
+
+    for (var subject : newInstance.getSubjects()) {
+      if (subject.getSourceId() == null) {
+        instanceRepository.unlinkInstanceFromSubjectSource(instanceId);
```

Comment:
Here you unlink all subjects from the instance, but should unlink only one based on your logic.

Reply:
here, i'm iterating over subjects and unlink if some of them does not exist in newInstance json